### CONFIDENCE INTERVALS

#### EXERCISE 1. 
What is the normal body temperature for healthy humans? A random sample of 130 healthy human body temperatures provided by Allen Shoemaker yielded 98.25 degrees and standard deviation 0.73 degrees. 

Give a 99% confidence interval for the average body temperature of healthy people.

In [19]:
import numpy as np
import pandas as pd
from scipy import stats

In [2]:
n = 130               # sample size
xbar = 98.25          # sample mean
s = 0.73              # sample std
sem = s / np.sqrt(n)  # standard error of the mean

In [3]:
stats.t.interval(confidence=0.99,  # confidence level  
                 df=n-1,           # degrees of freedom
                 loc=xbar,         # sample mean
                 scale=sem)        # standard error of the mean

(98.08260738705933, 98.41739261294067)

We are 99% confident that the average body temperature of healty people is between 98.08 and 98.41

#### EXERCISE 2. 
The administrators for a hospital wished to estimate the average number of days required for inpatient treatment of patients between the ages of 25 and 34. A random sample of 500 hospital patients between these ages produced a mean and standard deviation equal to 5.4 and 3.1 days, respectively.


Construct a 95% confidence interval for the mean length of stay for the population of patients from which the sample was drawn.

In [4]:
n = 500
xbar = 5.4
s = 3.1
sem = s / np.sqrt(n)

In [5]:
stats.t.interval(confidence=0.95,  # confidence level  
                 df=n-1,           # degrees of freedom
                 loc=xbar,         # sample mean
                 scale=sem)        # standard error of the mean

(5.127617354510309, 5.672382645489692)

We are 95% confident that the mean length of stay required for inpatient treatment of patients is between 5.12 and 5.67 days

### HYPOTHESIS TESTING

#### EXERCISE 3.
The hourly wages in a particular industry are normally distributed with mean \\$13.20 and standard deviation of \\$ 2.50. A company in this industry employs 40 workers, paying them an average of \$12.20 per hour. Can this company be accused of paying substandard wages? Use an α = .01 level test. (Wackerly, Ex.10.18)

CHECK: statistic: -2.5298221281347035, pvalue= 0.005706018193000826

In [6]:
# H0: mu = 13.20
# H1: mu < 13.20

In [7]:
n = 40
mu = 13.20
xbar = 12.20
sigma = 2.50
alpha = 0.01

In [8]:
se = sigma / np.sqrt(n)

In [11]:
z = (xbar - mu) / se
z

-2.5298221281347035

In [12]:
p_value = stats.norm.cdf(z)
p_value

0.005706018193000826

In [18]:
if p_value < alpha:
    print(f"At {alpha} level of significance level, there is enough evidence to support the claim that this company paying substandard wages")
else:
    print(f"At {alpha} level of significance level, there is not enough evidence to support the claim that this company paying substandard wages")

At 0.01 level of significance level, there is enough evidence to support the claim that this company paying substandard wages


#### EXERCISE 4.
Shear strength measurements derived from unconfined compression tests for two types of soils gave the results shown in the following document (measurements in tons per square foot). Do the soils appear to differ with respect to average shear strength, at the 1% significance level?

[Results for two type of soils](https://docs.google.com/spreadsheets/d/1f2odmgDboIVuSV-A5gmuC25ppqQ5g1OIIF4h5EOqUcI/edit?usp=sharing)

CHECK: statistic: 5.1681473319343345, pvalue= 2.593228732352821e-06

In [20]:
soil = pd.read_csv("soil.csv")

In [23]:
soil.head(10)

Soil1  Soil2
0  1.442  1.364
1  1.943  1.878
2  1.110  1.337
3  1.912  1.828
4  1.553  1.371
5  1.641  1.428
6  1.499  1.119
7  1.347  1.373
8  1.685  1.589
9  1.578  1.714

In [26]:
soil.shape

(35, 2)

In [24]:
soil.isna().sum()

Soil1    5
Soil2    0
dtype: int64

In [27]:
soil.dropna(inplace=True)

In [28]:
soil.shape

(30, 2)

In [29]:
#Perform Levene test for equal variances
#H0: The population variances are equal
#H1: There is a difference between the variances in the population
#The small p-value suggests that the populations do not have equal variances.

stats.levene(soil.Soil1, soil.Soil2)

LeveneResult(statistic=0.2323198108973329, pvalue=0.631622932753579)

 pvalue=0.6316 > alpha=0.01 ---> Fail to reject null hypothesis, the population variances are equal

In [30]:
indTest = stats.ttest_ind(soil.Soil1, soil.Soil2, equal_var=True)
indTest

Ttest_indResult(statistic=5.134893443609086, pvalue=3.4402046436336477e-06)

In [34]:
alpha = 0.01

if indTest.pvalue < alpha:
    print("Reject the null hypothesis")
else:
    print("Fail to reject null hypothesis")

Reject the null hypothesis


#### EXERCISE 5. 
The following dataset is based on data provided by the World Bank (https://datacatalog.worldbank.org/dataset/education-statistics). World Bank Edstats. [2015 PISA Test Dataset](https://docs.google.com/spreadsheets/d/14rVnIUfEm3CuK9bSvS5253RHWzQhXOuNc0I-cCkgpR8/edit?usp=sharing) 

1.Get descriptive statistics (the central tendency, dispersion and shape of a dataset’s distribution) for each continent group (AS, EU, AF, NA, SA, OC).
2.Determine whether there is any difference (on the average) for the math scores among European (EU) and Asian (AS) countries (assume normality and equal variances). Draw side-by-side box plots.

CHECK: statistic=0.870055317967983, pvalue=0.38826888111307345

In [36]:
df = pd.read_csv("2015_PISA_Test.csv")

In [37]:
df

Country Code Continent_Code  internet_users_per_100      Math   Reading  \
0           ALB             EU               63.252933  413.1570  405.2588   
1           ARE             AS               90.500000  427.4827  433.5423   
2           ARG             SA               68.043064  409.0333  425.3031   
3           AUS             OC               84.560519  493.8962  502.9006   
4           AUT             EU               83.940142  496.7423  484.8656   
..          ...            ...                     ...       ...       ...   
65          TUN             AF               48.519836  366.8180  361.0555   
66          TUR             EU               53.744979  420.4540  428.3351   
67          URY             SA               64.600000  417.9919  436.5721   
68          USA            NaN               74.554202  469.6285  496.9351   
69          VNM             AS               43.500000  494.5183  486.7738   

     Science  
0   427.2250  
1   436.7311  
2   432.2262  
3   509.9939  
4   495.0375  
..       ...  
65  386.4034  
66  425.4895  
67  435.3630  
68  496.2424  
69  524.6445  

[70 rows x 6 columns]

In [38]:
df.shape

(70, 6)

In [39]:
df.isna().sum()

Country Code              0
Continent_Code            5
internet_users_per_100    0
Math                      0
Reading                   0
Science                   0
dtype: int64

In [40]:
df.dropna(inplace=True)

In [41]:
df.shape

(65, 6)

In [55]:
df.groupby("Continent_Code").describe()

internet_users_per_100                                   \
                                count       mean        std        min   
Continent_Code                                                           
AF                                2.0  43.359918   7.297226  38.200000   
AS                               17.0  68.455613  21.086060  21.976068   
EU                               37.0  77.274888  12.425773  53.744979   
OC                                2.0  86.391704   2.589686  84.560519   
SA                                7.0  60.180494   9.772455  40.900000   

                                                            Math              \
                      25%        50%        75%        max count        mean   
Continent_Code                                                                 
AF              40.779959  43.359918  45.939877  48.519836   2.0  363.212100   
AS              50.300000  74.000000  84.948353  92.884826  17.0  466.216647   
EU              68.632900  76.184000  87.479056  98.200000  37.0  477.981449   
OC              85.476112  86.391704  87.307296  88.222889   2.0  494.559750   
SA              57.116462  64.289000  66.321532  69.198471   7.0  402.887700   

                ...     Reading           Science                         \
                ...         75%       max   count        mean        std   
Continent_Code  ...                                                        
AF              ...  358.256450  361.0555     2.0  381.074250   7.536556   
AS              ...  508.690500  535.1002    17.0  467.945847  56.671371   
EU              ...  499.814600  526.4247    37.0  478.299381  34.450616   
OC              ...  507.678175  509.2707     2.0  511.648700   2.340241   
SA              ...  431.922700  458.5709     7.0  421.747186  18.470319   

                                                                       
                     min         25%        50%         75%       max  
Continent_Code                                                         
AF              375.7451  378.409675  381.07425  383.738825  386.4034  
AS              386.4854  417.611200  456.48360  523.277400  555.5747  
EU              383.6824  460.774900  490.22500  501.936900  534.1937  
OC              509.9939  510.821300  511.64870  512.476100  513.3035  
SA              396.6836  408.205450  424.59050  433.794600  446.9561  

[5 rows x 32 columns]